chapters/shaders.adoc

```diff
@@ -2498,6 +2498,86 @@ In other shader stages, each invocation in a subgroup must: be in the same
 
 Only <<limits-subgroup-supportedStages, shader stages that support subgroup
 operations>> have defined subgroups.
+
+[NOTE]
+.Note
+====
+In shaders, there are two kinds of uniformity that is of interest to
+applications, subgroup uniform and dynamically uniform.
+
+While dynamically uniform appears to be a stronger kind of uniform
+than subgroups, it is not necessarily the case for graphics pipelines.
+
+For shader stages with defined workgroups, this assumption holds.
+If a value is dynamically uniform, it is by definition also uniform across the subgroup,
+as this is a specific guarantee provided to stages with explicit workgroups.
+This is important if writing code like:
+
+[source,c]
+~~~~
+uniform texture2D Textures[];
+uint dynamicallyUniformValue = gl_WorkGroupID.x;
+vec4 value = texelFetch(Textures[dynamicallyUniformValue], coord, 0);
+
+// subgroupUniformValue is guaranteed to be uniform across the subgroup.
+// This value also happens to be dynamically uniform.
+vec4 subgroupUniformValue = subgroupBroadcastFirst(dynamicallyUniformValue);
+~~~~
+
+In fragment shaders and other graphics stages, this gets complicated.
```

Comment:
Other graphics stages is once again not correct. I'm not sure what a precise and good wording would be. Perhaps enumerate all stages? This should also include RT

Reply:
Reworded.